### PR TITLE
Add script to validate remote object store paths

### PR DESCRIPTION
**Pull request by irenedea**
# What does this PR do?

Adds a script that validates remote object store paths using composer utility functions. 

This is helpful for debugging especially on remote runs. 

# What issue(s) does this change relate to?

# Before submitting
- [ ] Have you read the [contributor guidelines](https://github.com/mosaicml/composer/blob/dev/CONTRIBUTING.md)?
- [ ] Is this change a documentation change or typo fix? If so, skip the rest of this checklist.
- [ ] Was this change discussed/approved in a GitHub issue first? It is much more likely to be merged if so.
- [ ] Did you update any related docs and document your change?
- [ ] Did you update any related tests and add any new tests related to your change? (see [testing](https://github.com/mosaicml/composer/blob/dev/CONTRIBUTING.md#running-tests))
- [ ] Did you run the tests locally to make sure they pass?
- [ ] Did you run `pre-commit` on your change? (see the `pre-commit` section of [prerequisites](https://github.com/mosaicml/composer/blob/dev/CONTRIBUTING.md#prerequisites))

### CODE REVIEW

**Comment by j316chuck** [comment]
qq: Is this script supposed to be used locally?

**Review by j316chuck — APPROVED** [review]

**Comment by irenedea** [comment]
@j316chuck the mcli yaml that i was using only supports s3, so i figured this would be an easy way to check all backends easily and through composer as well. Though, if this script fails, it doesn't necessarily rule out composer issues. I'll write an mcli yaml for this as well.

**Review by dakinggg — CHANGES_REQUESTED** [review]
Can we make this part of the library instead of a script? similar to `composer_collect_env`

**Comment by irenedea** [comment]
@dakinggg good suggestion! so it would both in the library as well as a script, yeah?

**Comment by dakinggg** [comment]
Yeah, exact same as composer_collect_env setup

**Comment by irenedea** [comment]
@dakinggg I think the best move would be to create a `list_remote_objects` (that does exactly what the script does now, but returns a list of objects if objects are found). This would probably belong in `file_helpers.py`. If we add this function to `file_helpers.py`, do you have any thoughts on where to put the script?

**Comment by dakinggg** [comment]
Can you define `console_scripts` that don't use `main`? It kinda looks like you can, but I've never tried it. If so, can just define a function in `file_helpers.py` that prints out what you want, and define the `console_scripts` command to point to that function

**Comment by irenedea** [comment]
@dakinggg Tried it, but it won't work--it needs to a be a function that takes no arguments. So we could (1) keep a tiny script in `scripts/validate_remote_path.py` and have it call the `list_remote_objects` function in `file_helpers.py` (2) create a small function that takes no args that is defined in `file_helpers.py` that acts as  a console script

**Comment by dakinggg** [comment]
I think Its possible to set it up with args parsed by argparse. because the `composer` launcher is set up the same way. See the main function in there that parses args

File: setup.py [hunk 1]
```diff
@@ -264,8 +264,8 @@ def package_files(prefix: str, directory: str, extension: str):
       install_requires=install_requires,
       entry_points={
           'console_scripts': [
-              'composer = composer.cli.launcher:main',
-              'composer_collect_env = composer.utils.collect_env:main',
+              'composer = composer.cli.launcher:main', 'composer_collect_env = composer.utils.collect_env:main',
+              'composer_validate_remote_path = composer.utils.file_helpers:validate_remote_path'
```
Comment (irenedea) [inline setup.py:268]: validated that this works
```
composer_validate_remote_path oci://<path>
```

Comment (dakinggg) [inline setup.py:268]: add back trailing comma so this is formatted nicely

File: composer/utils/file_helpers.py [hunk 2]
```diff
@@ -400,6 +400,44 @@ def maybe_create_remote_uploader_downloader_from_uri(
...
+    _, _, prefix = parse_uri(remote_path)
+    objects = object_store.list_objects(prefix)
+    return objects
+
+
+def validate_remote_path():
+    """Entry point to composer_validate_remote_path cli command.
+
+    Validates a remote path.
+    If a the remote path is valid, prints a list of objects at the path.
```
Comment (dakinggg) [inline composer/utils/file_helpers.py:424]:
```suggestion
    If the remote path is valid, prints a list of objects at the path.
```

**Review by dakinggg — APPROVED** [review]

**Pull request merged**